src/drawing/DrawingAnnotations.tsx

```diff
@@ -1,30 +1,86 @@
 import * as React from 'react';
+import * as uuid from 'uuid';
 import DrawingList from './DrawingList';
-import { AnnotationDrawing } from '../@types';
+import DrawingCreator from './DrawingCreator';
+import DrawingPathGroup from './DrawingPathGroup';
+import DrawingSVG, { DrawingSVGRef } from './DrawingSVG';
+import { AnnotationDrawing, PathGroup } from '../@types';
+import { CreatorItemDrawing, CreatorStatus } from '../store';
 import './DrawingAnnotations.scss';
 
 export type Props = {
     activeAnnotationId: string | null;
     annotations: AnnotationDrawing[];
+    isCreating: boolean;
     isCurrentFileVersion: boolean;
+    location: number;
     setActiveAnnotationId: (annotationId: string | null) => void;
+    setStaged: (staged: CreatorItemDrawing | null) => void;
+    setStatus: (status: CreatorStatus) => void;
+    staged?: CreatorItemDrawing | null;
 };
 
 const DrawingAnnotations = (props: Props): JSX.Element => {
-    const { activeAnnotationId, annotations, isCurrentFileVersion, setActiveAnnotationId } = props;
+    const {
+        activeAnnotationId,
+        annotations,
+        isCreating,
+        isCurrentFileVersion,
+        location,
+        setActiveAnnotationId,
+        setStaged,
+        setStatus,
+        staged,
+    } = props;
+    const [stagedRootEl, setStagedRootEl] = React.useState<DrawingSVGRef | null>(null);
+    const stagedGroupRef = React.useRef<SVGGElement>(null);
+    const uuidRef = React.useRef<string>(uuid.v4());
 
     const handleAnnotationActive = (annotationId: string | null): void => {
         setActiveAnnotationId(annotationId);
     };
+    const handleStart = (): void => {
+        if (staged === null) {
+            setStaged(null);
+        }
+        setStatus(CreatorStatus.started);
+    };
+    const handleStop = (pathGroup: PathGroup): void => {
+        const { drawnPathGroups = [] } = staged || {};
+        const payload = {
+            drawnPathGroups: drawnPathGroups.concat(pathGroup),
+            location,
+            stashedPathGroups: [],
+        };
+
+        setStaged(payload);
```

Comment:
Inline `payload`?

Reply:
This should be resolved as part of @jstoffan 's comment above https://github.com/box/box-annotations/pull/640/files#r532852562